src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,182 @@
+package core.basesyntax.service;
+
+import core.basesyntax.InvalidDataException;
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private static RegistrationServiceImpl registrationService;
+    private User user1;
+    private User user2;
+    private User user3;
+    private User user4;
+
+    @BeforeAll
+    static void beforeAll() {
+        registrationService = new RegistrationServiceImpl();
+    }
+
+    @BeforeEach
+    void setUp() {
+        Storage.people.clear();
+
+        user1 = new User();
+        user1.setAge(18);
+        user1.setLogin("username1");
+        user1.setPassword("123456");
+
+        user2 = new User();
+        user2.setAge(18);
+        user2.setLogin("username2");
+        user2.setPassword("123456");
+
+        user3 = new User();
+        user3.setAge(18);
+        user3.setLogin("username3");
+        user3.setPassword("123456");
+
+        user4 = new User();
+        user4.setAge(18);
+        user4.setLogin("username4");
+        user4.setPassword("123456");
+    }
+
+    @Test
+    void register_nullAge_notOk() {
+        user1.setAge(null);
+        Assertions.assertThrows(InvalidDataException.class,
+                () -> registrationService.register(user1));
+    }
+
+    @Test
+    void register_negativeAge_notOk() {
+        user1.setAge(Integer.MIN_VALUE);
+        Assertions.assertThrows(InvalidDataException.class,
+                () -> registrationService.register(user1));
+        user1.setAge(-10);
+        Assertions.assertThrows(InvalidDataException.class,
+                () -> registrationService.register(user1));
+        user1.setAge(-1);
+        Assertions.assertThrows(InvalidDataException.class,
+                () -> registrationService.register(user1));
+    }
+
+    @Test
+    void register_insufficientAge_notOk() {
```

Comment:
```suggestion
    void register_underage_notOk() {
```
we call ppl who haven't reached 18 underage